### PR TITLE
Clarify/fix `proj:epsg` requirement

### LINKED ISSUE #7: `proj:epsg` is marked as "required", but this only applies to items
The readme says [`proj:epsg` is required](https://github.com/stac-extensions/projection/blob/7f1d1c9db7c33dba0944972a983d74fd480d4c62/README.md#L31), but [it's only actually required inside items](https://github.com/stac-extensions/projection/blob/7f1d1c9db7c33dba0944972a983d74fd480d4c62/json-schema/schema.json#L26). Is this a bug in the schema, a missing piece of documentation, or have I misunderstood something?

### LINKED ISSUE #16: Best pratice: Provide proj:shape for thumbnails
For browser rendering it would be very beneficial to provide `proj:shape` also to thumbnails (and overviews) so that image dimensions are known before loading them: 
```
{
  "href": "thumb.png",
  "title": "Thumbnail",
  "type": "image/png",
  "roles": ["thumbnail"],
  "proj:shape": [400, 400],
  "proj:epsg": null
}
```

See also: https://github.com/radiantearth/stac-browser/issues/166

### PR DESCRIPTION
**Pull request by m-mohr**
Fixes #7 + #16

This requires `proj:epsg` in the scope where `proj` fields are used. 

This might be an issue for cases where currently proj:epsg is in Item properties and the individual assets use proj:shape/proj:transform only. I think I've seen that in implementations...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +19,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Added missing scope "Collection" to Readme. The scope was already supported in JSON Schemas.
+- Clarified that `proj:epsg` is not required in Item properties, except that another proj field is used there.
+  In general, `proj:epsg` is required in the scope where other proj fields are used.
```
Comment (hobu) [inline CHANGELOG.md]: Is this actually true? I would say that the `proj:epsg` field is the most frequently referenced and used. A frustrating consequence of this change is now all software that consumes `proj` extensions is essentially required to consume *all* of the definition types. For the Python ecosystem that has pyproj/PROJ, this is fine, but for others that do not have it, it is not so great.

File: README.md [hunk 2]
```diff
@@ -37,6 +46,9 @@ The `proj` prefix is short for "projection", and is not a reference to the PROJ/
 | proj:shape       | \[integer]      | Number of pixels in Y and X directions for the default grid |
 | proj:transform   | \[number]       | The affine transformation coefficients for the default grid  |
 
+If any of the fields is used, `proj:epsg` is required in the scope where the field is used.
+For example, if you use `proj:shape` is used in an Item Asset, you need to specify `proj:epsg` in that specific Asset, too.
```
Comment (hobu) [inline README.md]: This language is confusing. Are we now saying that `proj:epsg` is required for any Item that has a `proj:wkt2`? We can't reliably do that for all WKT2, of course. I think this section needs to be rewritten to clarify intention and prescribe to users what they should do for various scenarios.

Comment (m-mohr) [reply README.md]: Yes, but you can set it to null as described in the README. This was already true beforehand, `proj:epsg` was always required to be set (EPSG code or null).

File: CHANGELOG.md (hunk 1 above)
Comment (m-mohr) [reply CHANGELOG.md]: I can't follow. What has changed compared to beforehand? Right now we just move the requirement from Item properties to the scope where the other proj fields are actually used and as such it looks like your concerns would have already been true before the PR...

Comment (hobu) [reply CHANGELOG.md]: Not so much a concern but rather me pointing out an implicit consequence of the current language.

Comment (m-mohr) [reply CHANGELOG.md]: I don't think there's a "functional" change here, we just change where the fields is required (asset vs. item). But yeah, something we can discuss in the STAC call today in a broader group...

**Comment by m-mohr** [comment]
Dev call: Don't require proj:epsg at all (but say that having EPSG would be great), see #7 for more discussions. Thus, the schema in this PR needs an update.

File: CHANGELOG.md (hunk 1 above)
Comment (m-mohr) [reply CHANGELOG.md]: Changed to not require proj:epsg any longer.

File: README.md (hunk 2 above)
Comment (m-mohr) [reply README.md]: Changed to not require proj:epsg any longer.

**Comment by m-mohr** [comment]
As discussed in a STAC meeting last year, we are removing the proj:epsg requirement. Updated this PR to reflect that, I'd appreciate a new round of reviews.

**Review by emmanuelmathot — COMMENTED** [review]
This is fine with me.

File: README.md [hunk 3]
```diff
@@ -186,6 +208,20 @@ WKT2 and PROJJSON are equivalent to one another - more clients understand WKT2,
 structure, since they are both JSON. For now it's probably best to use both for maximum interoperability, but just using PROJJSON 
 is likely ok if you aren't worried about legacy client support.
 
+### Thumbnails
+
+For (unlocated) thumbnails and similar imagery, it is recommended to add `proj:shape` with `proj:epsg` set to `null`
```
Comment (gadomski) [inline README.md]:
```suggestion
For (unlocated) thumbnails and similar imagery, it is recommended set `proj:epsg` to `null` and include `proj:shape`
```

Slight wordsmithing to make it clear that `proj:shape` and `proj:epsg` are not linked. To me, 

> add `proj:shape` with `proj:epsg`

scans like `proj:epsg` is a component of `proj:shape`.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (m-mohr) [reply README.md]: Good point, updated.

**Review by gadomski — APPROVED** [review]

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**